Code/Framework/AzCore/AzCore/Serialization/EditContext.h

```diff
@@ -16,120 +17,168 @@
+        const char* m_deprecatedName = nullptr;
+        SerializeContext::ClassElement* m_serializeClassElement; ///< If nullptr this is class (logical) element, not physical element exists in the class
+        AttributeArray      m_attributes;
+    };
 
-        using Attribute = AZ::Attribute;
-        template<class T>
-        using AttributeData = AZ::AttributeData<T>;
-        template<class T>
-        using AttributeMemberData = AZ::AttributeMemberData<T>;
-        template<class F>
-        using AttributeFunction = AZ::AttributeFunction<F>;
-        template<class F>
-        using AttributeMemberFunction = AZ::AttributeMemberFunction<F>;
+    /**
+     * Class data is assigned to every Serialize::Class. Don't confuse m_elements with
+     * ElementData. Elements contains class Elements (like groups, etc.) while the ElementData
+     * contains attributes related to ehe SerializeContext::ClassBuilder::Field.
+     */
+    struct ClassData
+    {
+        ClassData()
+            : m_name(nullptr)
+            , m_description(nullptr)
+            , m_classData(nullptr)
+            , m_editDataProvider(nullptr)
+        {}
+
+        void ClearElements();
+        const ElementData* FindElementData(AttributeId elementId) const;
+
+        const char* m_name;
+        const char* m_description;
+        SerializeContext::ClassData* m_classData;
+        DynamicEditDataProvider* m_editDataProvider;
+        AZStd::list<ElementData>    m_elements;
+    };
+} // namespace AZ::Edit
 
-        /**
-         * Enumerates the serialization context and inserts component Uuids which match at least one tag in requiredTags
-         * for the AZ::Edit::SystemComponentsTag attribute on the reflected class data
-         */
-        void GetComponentUuidsWithSystemComponentTag(
-            const SerializeContext* serializeContext,
-            const AZStd::vector<AZ::Crc32>& requiredTags,
-            AZStd::unordered_set<AZ::Uuid>& componentUuids);
+namespace AZ::SerializeInternal
+{
+    template<class T>
+    struct ElementInfo;
 
-        /**
-         * Returns true if the given classData's AZ::Edit::SystemComponentsTag attribute matches at least one tag in requiredTags
-         * If the class data does not have the attribute, returns the value of defaultVal.
-         * Note that a defaultVal of true should generally only be used to maintain legacy behavior with non-tagged AZ::Components
-         */
-        template <typename TagContainer>
-        bool SystemComponentTagsMatchesAtLeastOneTag(
-            const AZ::SerializeContext::ClassData* classData,
-            const TagContainer& requiredTags,
-            bool defaultVal = false);
+    template<class T, class C>
+    struct ElementInfo<T C::*>
+    {
+        using ElementType = AZStd::RemoveEnumT<T>;
+        using ClassType = C;
+        using Type = T;
+        using ValueType = AZStd::remove_pointer_t<ElementType>;
+    };
+} // namespace AZ::SerializeInternal
```

Comment:
This code wasn't changed.
It was just updated to use the C++17 nested namespace feature.